### PR TITLE
Add UnnecessaryLocalRule to detect assign-then-return pattern

### LINKED ISSUE #102: Implement UnnecessaryLocalRule
Forbid local variables that are assigned and immediately returned or thrown.

## Scope

- Method body analysis for assign-then-return/throw pattern

## Conditions

- variable assigned and immediately returned on next statement triggers error
- variable assigned and immediately thrown on next statement triggers error
- variable used elsewhere in method besides return/throw does not trigger error
- variable with `@var` PHPDoc annotation does not trigger error
- assignments inside closures, arrow functions, anonymous classes are not checked

## Expected Result

- Unnecessary local variables before return/throw are reported
- Legitimate uses of intermediate variables are not affected

### PR DESCRIPTION
**Pull request by haspadar**
## Summary
- Add `UnnecessaryLocalRule` that reports local variables assigned and immediately returned or thrown
- Variables with `@var` PHPDoc are excluded (they carry type information)
- Register rule in `rules.neon`, add to README, add tests and fixtures

Closes #102

## Summary by CodeRabbit

* **New Features**
  * Added a PHPStan rule that flags local variables assigned and immediately returned or thrown, recommending direct return/throw.

* **Documentation**
  * Updated README with the new rule entry and description.

* **Tests**
  * Added comprehensive fixtures and unit tests covering valid, failing, suppressed, and edge-case scenarios to validate the rule.

### CODE REVIEW

File: src/Rules/UnnecessaryLocalRule.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        }
+
+        if ($node->stmts === null) {
+            return [];
+        }
+
+        $errors = [];
+
+        foreach ($this->findUnnecessaryPairs(array_values($node->stmts)) as [$name, $line, $kind]) {
+            $errors[] = RuleErrorBuilder::message(
```
[the remark this answers is not in the document]
Comment (haspadar) [reply src/Rules/UnnecessaryLocalRule.php:59]: By design. The rule intentionally checks only the flat top-level statement list of a method, matching the behavior of PMD `VariableCanBeInlined`, SonarQube `S1488`, and PHP-CS-Fixer `return_assignment`.

Nested blocks (`if`/`else`, `try`/`catch`) are excluded because:
- The variable may serve scope or cleanup purposes inside blocks
- Analyzing nested blocks creates false positives with `finally`, early returns, etc.

Regarding "over-report reused locals": the rule only fires when the **immediately next** statement is `return $var` or `throw $var`. If anything else appears between the assignment and the return (like `$this->log($result)`), the next statement check fails and no error is reported. The `UsedElsewhere` fixture validates this.

Added `InsideIfBlock` fixture to explicitly document nested block behavior.
